tests/system/python/packages/test_statistics_history_notification_rule.py

```diff
@@ -0,0 +1,295 @@
+                on endpoint GET /fledge/ping
+                on endpoint GET /fledge/statistics/history """
+        # Change the config of threshold, name of statistics - READINGS replaced with statistics key name - Sine #1-Ingest
+        put_url = "/fledge/category/ruletest #1"
+        data = {"asset": "Sine #1-Ingest"}
+        utils.put_request(fledge_url, urllib.parse.quote(put_url), data)
+        
+        verify_ping(fledge_url, skip_verify_north_interface, wait_time, retries)
+        
+        # When rule is triggered, there should be audit entries for NTFSN
+        get_url = "/fledge/audit?source=NTFSN"
+        resp1 = utils.get_request(fledge_url, get_url)
+        assert len(resp1['audit'])
+        assert "test #1" in [s["details"]["name"] for s in resp1["audit"]]
+        for audit_detail in resp1['audit']:
+            if "test #1" == audit_detail['details']['name']:
+                assert "NTFSN" == audit_detail['source']
+        # Waiting for 60 sec to get more NTFSN entries
+        time.sleep(60)
+        resp2 = utils.get_request(fledge_url, get_url)
+        assert len(resp2['audit']) - len(resp1['audit']) == 2, "ERROR: NTFSN not triggered properly"
+        
+        get_url = "/fledge/statistics/history?minutes=10"
+        r = utils.get_request(fledge_url, get_url)
+        if "Sine #1-Ingest" in r["statistics"][0]:
+            assert 0 < r["statistics"][0]["Sine #1-Ingest"]
+        
+    def test_stats_south_asset(self, fledge_url, wait_time, skip_verify_north_interface, retries):
+        """ Test NTFSN triggered or not with source as statistics history and name as south asset name in threshold rule.
+            Assertions:
+                on endpoint GET /fledge/audit
+                on endpoint GET /fledge/ping
+                on endpoint GET /fledge/statistics/history """
+        # Change the config of threshold, name of statistics - Sine #1-Ingest replaced with statistics key name - 20230420_SINUSOID_ASSETS
+        put_url = "/fledge/category/ruletest #1"
+        data = {"asset": SOUTH_ASSET_NAME.upper()}
+        utils.put_request(fledge_url, urllib.parse.quote(put_url), data)
+        
+        verify_ping(fledge_url, skip_verify_north_interface, wait_time, retries)
+        
+        # When rule is triggered, there should be audit entries for NTFSN
+        get_url = "/fledge/audit?source=NTFSN"
+        resp1 = utils.get_request(fledge_url, get_url)
+        assert len(resp1['audit'])
+        assert "test #1" in [s["details"]["name"] for s in resp1["audit"]]
+        for audit_detail in resp1['audit']:
+            if "test #1" == audit_detail['details']['name']:
+                assert "NTFSN" == audit_detail['source']
+        # Waiting for 60 sec to get more NTFSN entries
+        time.sleep(60)
+        resp2 = utils.get_request(fledge_url, get_url)
+        assert len(resp2['audit']) - len(resp1['audit']) == 2, "ERROR: NTFSN not triggered properly"
```

Comment:
I have tried executing this test multiple time in my local machine but this test failed in every runof this pytest at this step.
Please share the link of any Jenkins job that is successful.